src/lib/ares_process.c

```diff
@@ -147,6 +155,17 @@ static void server_set_good(ares_server_t *server, ares_bool_t used_tcp)
 
   if (server->consec_failures > 0) {
     server->consec_failures = 0;
```

Comment:
should not unset consec_failures until consec_successes passes threshold.  again, we don't want to track with yet another variable of is_failed.